lib/common/common/src/iterator_ext/mod.rs

```diff
@@ -26,6 +28,20 @@ pub trait IteratorExt: Iterator {
     {
         self.check_stop_every(500, f)
     }
+
+    /// Will execute the callback when the iterator is dropped.
+    ///
+    /// The callback receives the total number of times `.next()` was called on the iterator,
+    /// including the final one where it usually returns `None`.
+    ///
+    /// Consider subtracting 1 if the final `None` is not needed.
+    fn on_final_count<F>(self, f: F) -> OnFinalCount<Self, F>
```

Comment:
A bit verbose, but how about `count_callback_on_drop`? It might give a better hint on what it does without looking at the function.

I've no strong opinion on either.

cc @JojiiOfficial

Reply:
I'm fine with both but agree this being more self-explanatory